### ISSUE TITLE
Panic after a few seconds running - version 1.5.4 (Home Assistant Add On)

### ISSUE THREAD

**Issue by john94501** [mrlt8/docker-wyze-bridge#419]
After a few seconds running (in Home Assistant), I get the following in the logs:

```
🚀 STARTING DOCKER-WYZE-BRIDGE v1.5.4
🏠 Home Assistant Mode
2022/06/14 13:58:00 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'user' data
2022/06/14 13:58:00 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'auth' data
2022/06/14 13:58:00 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2022/06/14 13:58:00 [WyzeBridge] 💾 Saving 'auth' to local cache...
2022/06/14 13:58:00 [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2022/06/14 13:58:00 [WyzeBridge] 💾 Saving 'user' to local cache...
2022/06/14 13:58:00 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'cameras' data
2022/06/14 13:58:00 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2022/06/14 13:58:01 [WyzeBridge] 💾 Saving 'cameras' to local cache...
🎬 STARTING ALL 4 CAMERAS
2022/06/14 13:58:01 [WyzeBridge] Starting rtsp-simple-server v0.19.1
2022/06/14 13:58:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - Floodlight on 192.168.7.105 (1/3)
2022/06/14 13:58:01 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Office Cam on 192.168.7.110 (1/3)
2022/06/14 13:58:01 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Front Door on 192.168.7.238 (1/3)
2022/06/14 13:58:01 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Kitchen on 192.168.7.237 (1/3)
2022/06/14 13:58:03 [Office Cam] 📡 Getting 120kb/s HD stream (15fps) via LAN mode (WiFi: 98%) FW: 4.9.8.1002 🔒 (DTLS) (2/3)
2022/06/14 13:58:03 [Front Door] 📡 Getting 120kb/s HD stream (15fps) via LAN mode (WiFi: 88%) FW: 4.9.8.1002 🔒 (DTLS) (2/3)
2022/06/14 13:58:03 [RTSP][OFFICE-CAM] ✅ '/office-cam' stream is UP! (3/3)
2022/06/14 13:58:03 [Floodlight] 📡 Getting 120kb/s HD stream (20fps) via LAN mode (WiFi: 38%) FW: 4.36.9.139 🔒 (DTLS) (2/3)
2022/06/14 13:58:03 [Floodlight] WARNING: Skipping smaller frame at start of stream (frame_size=1)
2022/06/14 13:58:03 [Kitchen] 📡 Getting 120kb/s HD stream (15fps) via LAN mode (WiFi: 94%) FW: 4.9.8.1002 🔒 (DTLS) (2/3)
2022/06/14 13:58:03 [RTSP][FRONT-DOOR] ✅ '/front-door' stream is UP! (3/3)
2022/06/14 13:58:04 [RTSP][KITCHEN] ✅ '/kitchen' stream is UP! (3/3)
2022/06/14 13:58:04 [RTSP][FLOODLIGHT] ✅ '/floodlight' stream is UP! (3/3)
2022/06/14 13:58:08 [RTSP][FLOODLIGHT] 📖 New client reading
2022/06/14 13:58:15 [Kitchen] WARNING: Frame not available yet
2022/06/14 13:58:18 [Floodlight] WARNING: Frame not available yet
2022/06/14 13:58:20 [Front Door] WARNING: Frame not available yet
2022/06/14 13:58:27 [Office Cam] WARNING: Frame not available yet
2022/06/14 13:58:33 [RTSP][KITCHEN] 📖 New client reading
2022/06/14 13:58:37 [RTSP][KITCHEN] 📖 New client reading
2022/06/14 13:58:41 [RTSP][KITCHEN] 📕 Client stopped reading
2022/06/14 13:58:41 [RTSP][KITCHEN] 📕 Client stopped reading
2022/06/14 13:58:41 [RTSP][FLOODLIGHT] 📕 Client stopped reading
2022/06/14 13:58:45 [RTSP][KITCHEN] 📖 New client reading
2022/06/14 13:58:46 [RTSP][FLOODLIGHT] ❌ '/floodlight' stream is down
panic: assignment to entry in nil map
goroutine 329 [running]:
github.com/aler9/gortsplib.(*ServerStream).readerAdd(0x27cc780, 0x27fc090, 0x2, 0x0)
        /go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serverstream.go:175 +0x3b8
github.com/aler9/gortsplib.(*ServerSession).handleRequest(0x27fc090, 0x2668030, 0x268c000)
        /go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:706 +0x19ec
github.com/aler9/gortsplib.(*ServerSession).runInner(0x27fc090)
        /go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:342 +0x26c
github.com/aler9/gortsplib.(*ServerSession).run(0x27fc090)
        /go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:272 +0x110
created by github.com/aler9/gortsplib.newServerSession
        /go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:217 +0x298
[tee @ 0x24ef5f0] All tee outputs failed.
av_interleaved_write_frame(): Broken pipe
2022/06/14 13:58:46 [Floodlight] Stream did not receive a frame for over 20s
2022/06/14 13:58:47 [Office Cam] [Errno 32] Broken pipe
2022/06/14 13:58:47 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - Floodlight on 192.168.7.105 (1/3)
2022/06/14 13:58:47 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Office Cam on 192.168.7.110 (1/3)
[tee @ 0x3686410] All tee outputs failed.
av_interleaved_write_frame(): Broken pipe

```

After that none of the camera feeds are working.

**Comment by john94501** [comment]
More information. I fired it up on a PC running under Docker (so no Home Assistant). It ran for a while, and then suddenly failed like this:

```
Fatal Python error: init_import_site: Failed to import the site module
Python runtime state: initialized
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site.py", line 617, in <module>
    main()
  File "/usr/local/lib/python3.10/site.py", line 610, in main
    execsitecustomize()
  File "/usr/local/lib/python3.10/site.py", line 549, in execsitecustomize
    import sitecustomize
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1002, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 945, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1439, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1411, in _get_spec
  File "<frozen importlib._bootstrap_external>", line 1544, in find_spec
  File "<frozen importlib._bootstrap_external>", line 147, in _path_stat
KeyboardInterrupt
2022/06/15 00:32:52 [RTSP][FLOODLIGHT] ❌ '/floodlight' stream is down
panic: assignment to entry in nil map

goroutine 1078 [running]:
github.com/aler9/gortsplib.(*ServerStream).readerAdd(0xc0007f0580, 0xc000394400, 0x2, 0x0)
	/go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serverstream.go:175 +0x3fd
github.com/aler9/gortsplib.(*ServerSession).handleRequest(0xc000394400, 0xc0007fac00, 0xc0004cd2c0)
	/go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:706 +0x1872
github.com/aler9/gortsplib.(*ServerSession).runInner(0xc000394400)
	/go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:342 +0x32e
github.com/aler9/gortsplib.(*ServerSession).run(0xc000394400)
	/go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:272 +0x128
created by github.com/aler9/gortsplib.newServerSession
	/go/pkg/mod/github.com/aler9/gortsplib@v0.0.0-20220604232719-2fa3148a2751/serversession.go:217 +0x2bf
[tee @ 0x559a7d3c9440] All tee outputs failed.
av_interleaved_write_frame(): Broken pipe
2022/06/15 00:32:53 [Floodlight] Stream did not receive a frame for over 20s
2022/06/15 00:32:53 [Kitchen] [Errno 32] Broken pipe
[tee @ 0x562f5ffe4d40] All tee outputs failed.

```

**Comment by mrlt8** [comment]
Thanks for the logs. This seems to be a known issue in the upstream rtsp-simple-server.

Do you still get the error if filtering out some of the cameras with `FILTER_NAMES` e.g., `FILTER_NAMES=kitchen, front door`

**Comment by john94501** [comment]
Tried with two, and it still failed the same way after ~15 minutes. Will try with just one - the V3 camera that I added most recently - and see if that is the same. If so, I can retest with just one of the older V2 cams.

**Comment by john94501** [comment]
Was able to get it to happen with just the one V3 cam, but then today I have had it running with three cameras, including the V3 one, and it has been up for hours (with an application connected and consuming two of the feeds the whole time, including the feed from the V3).

**Comment by mrlt8** [comment]
look like there's a fix for this in gortsplib:
https://github.com/aler9/gortsplib/commit/aa08f973ddb8e5a1a18eecec4b60bb7d29396930

that should get pulled into the bridge when it's merged upstream.

**Comment by mrlt8** [comment]
@john94501 fix should be merged into the latest v1.6.6!

**Comment by john94501** [comment]
Interestingly, it hasn't happened for a few days now, but I will update immediately. Thanks for letting me know.
